docs env variable to control preview environments (#4109)
mesosphere/dcos-docs-site: README.md

If you need more control, have a look at the Makefile and the target `docker-liveedit`.

## Production and preview domains, DOCS_ENV variable

`main` is the default branch for this repo.

D2iQ has multiple domains for documentation to support various use cases: production, beta, and development. The config.json file can be configured to hide certain directories (that may be a work in progress) from being visible on each site.

- https://docs.d2iq.com/ is our production URL and will deploy when the `main` branch is changed. It receives the env variable `DOCS_ENV=production`.
- https://beta-docs.d2iq.com/ is our password protected beta URL and will deploy when the `beta` branch is changed. It receives the env variable `DOCS_ENV=beta`.
- https://dev-docs.d2iq.com/ is a preview domain and will deploy when the `main` branch is changed. It receives the env variable `DOCS_ENV=preview`.
- `"docs-d2iq-com-pr-${env.CHANGE_ID}.s3-website-us-west-2.amazonaws.com"` domains will deploy on every other branch. They receive the env variable `DOCS_ENV=preview`.

### Hiding directories based on DOCS_ENV

> To prevent a directory from appearing on the docs site in a specific domain, change the `config.json` like so:

Reviewer: ```suggestion
To prevent a directory from appearing on the docs site in a specific domain, change the `config.json` as follows:
```

```json
{
  "main": {
    "DO_NOT_BUILD": [
      "dkp/konvoy/42.0/**"
    ]
  }
}
```

> This code would tell Metalsmith to not build the Konvoy 42.0 section and its child pages.

Reviewer: ```suggestion
This code instructs Metalsmith not to build the Konvoy 42.0 section and its child pages.
```
